### PR TITLE
Prevent built-in-scripts from being made from FileSystem dock

### LINKED ISSUE #20350: Built-In Scripts Should Not be allowed from FileSystem Dock
**Godot version:**
latest_master custom build

**Issue description:**
When you create a script from the FileSystem Dock that is built-in there is no way to access it so why even allow it.
I have yet to see the code for Built-In Scripts so, and as a curious question what happens when we create a Built-In Script this way, is it instantly cleared from memory or does it just hold space in the current scene.
image
image

**Solution Ideas**
I think the easiest will be to create a new Dialog along the lines of `ScriptCreateDialog` but without the built-in option. Cause `ScriptCreateDialog` is used in other places as well. https://github.com/godotengine/godot/blob/525384d8cc45a0976e0afb862c8a54244bbd72e7/editor/filesystem_dock.h#L133

### PR DESCRIPTION
**Pull request by Piet-G**
Prevent built-in-scripts from being made from FileSystem dock.

Fixes #20350

### CODE REVIEW

**Comment by groud** [comment]
I'm not sure it worth creating a function just for such simple check. ^^ But in any case, the PR is fine.

File: editor/script_create_dialog.cpp [hunk 1]
```diff
@@ -588,7 +594,7 @@ void ScriptCreateDialog::_bind_methods() {
 	ClassDB::bind_method("_path_entered", &ScriptCreateDialog::_path_entered);
 	ClassDB::bind_method("_template_changed", &ScriptCreateDialog::_template_changed);
 
-	ClassDB::bind_method(D_METHOD("config", "inherits", "path"), &ScriptCreateDialog::config);
+	ClassDB::bind_method(D_METHOD("config", "inherits", "path", "built_in_enabled"), &ScriptCreateDialog::config);
```
Comment (akien-mga) [inline editor/script_create_dialog.cpp]: You should add `DEFVAL(true)` at the end too.

Comment (DualMatrix) [reply editor/script_create_dialog.cpp]: Added that.

**Comment by DualMatrix** [comment]
Is there a reason why AppVeyor failed?

**Comment by DualMatrix** [comment]
Ah great, then this seems good to go.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**